src/Db/Schema/VersionAbstract.php

```diff
@@ -7,32 +7,68 @@
 use Neighborhoods\Kojo\Doctrine;
 use Neighborhoods\Kojo\Doctrine\Connection\DecoratorInterface;
 use Neighborhoods\Pylon\Data\Property\Defensive;
+use Doctrine\DBAL\Types\Type;
 
 abstract class VersionAbstract implements VersionInterface
 {
     use Defensive\AwareTrait;
     use Doctrine\Connection\Decorator\Repository\AwareTrait;
     protected $_createTable;
     protected $_tableName;
+    protected $_mostRecentUnsupportedType;
 
     public function applySchemaSetupChanges(): VersionInterface
     {
         $connection = $this->_getDoctrineConnectionDecoratorRepository()->getConnection(DecoratorInterface::ID_SCHEMA);
         if (!$connection->getSchemaManager()->tablesExist([$this->_getTableName()])) {
-            $connection->beginTransaction();
             try {
-                $this->_assembleSchemaChanges();
-                $connection->getSchemaManager()->createTable($this->_getCreateTable());
-                $connection->commit();
+                $connection->beginTransaction();
+                while (!$this->_canAssembleSchemaChanges()) {
+                    $connection->rollBack();
+                    $connection
+                        ->getDatabasePlatform()
+                        ->registerDoctrineTypeMapping(
+                            $this->_getMostRecentUnsupportedType(),
+                            Type::STRING
+                        );
+                    $connection->beginTransaction();
+                }
             } catch (\Throwable $throwable) {
                 $connection->rollBack();
                 throw $throwable;
             }
         }
-
         return $this;
     }
 
+    protected function _canAssembleSchemaChanges(): bool
+    {
+        try {
+            $this->_assembleSchemaChanges();
+        } catch (\Doctrine\DBAL\DBALException $e) {
+            if ($this->_isUnsupportedTypeException($e)) {
+                return false;
```

Comment:
Can we have a single return point please?

Reply:
Any particular reason? Introducing a local variable complicates the flow of logic IMO:

```php
        $hasUnsupportedType = false;

        try {
            $this->_assembleSchemaChanges();
        } catch (\Doctrine\DBAL\DBALException $e) {
            if (!$this->_isUnsupportedTypeException($e)) {
                throw $e;
            }

            $hasUnsupportedType = true;
        }

        return $hasUnsupportedType;